avocado/core/test.py

```diff
@@ -383,6 +384,11 @@ def _start_logging(self):
                                                        formatter,
                                                        self._ssh_logfile)
 
+        if isinstance(sys.stdout, output.LoggingFile):
+            sys.stdout.add_logger(logging.getLogger("avocado.test.stdout"))
+        if isinstance(sys.stderr, output.LoggingFile):
+            sys.stderr.add_logger(logging.getLogger("avocado.test.stderr"))
```

Comment:
Well if the reproducer was to use `avocado.test.stderr` before `file_handler` was added to handlers, than by removing it we'd get the same state. The problem is that simple write to `avocado.test.stderr` won't cause the bug so I'm not sure whether removing the handler breaks things.

Anyway the first commit should address the issue with all handlers removed, so it's not really necessary. One the other hand with the first commit it's not necessary to add the `avocado.test.*` handlers after we populate the handlers. That's why I'd go with the first commit only, because this adds complexity (in terms of code readability, not speed) without any actual results.

My problem is that the first commit is a workaround, which avoids this problem by ignoring messages, but it covers all possible issues. The second commit (this) is theoretically better as it does the right thing, but it's harder to follow the initialization process.